### ISSUE TITLE
Confusion regarding signs of in-context vectors

### ISSUE THREAD

**Issue by zhxieml** [shengliu66/ICV#2]
Congratulations on your excellent work! The idea is truly inspiring.

I have some questions about the calculation of ICV. In the original paper, Equation 1 uses $y - x$ (where $y$ is the target). However, in the [notebook](https://github.com/shengliu66/ICV/blob/ae4fb0a2bbbb4e468053e9c357a42efda4990e18/demo_notebook.ipynb), it seems to be the opposite. Does this mean it is moving in the opposite direction? Or does it have something to do with [svp_flip](https://github.com/shengliu66/ICV/blob/ae4fb0a2bbbb4e468053e9c357a42efda4990e18/utils/pca.py#L6)?

Thank you for clarifying this.

**Comment by shengliu66** [comment]
Hey,

Thanks for your attention to our work. Yes, in the Jupyter Notebook we use $x-y$, but when we generate the ICV, we also take the opposite [sign](https://github.com/shengliu66/ICV/blob/ae4fb0a2bbbb4e468053e9c357a42efda4990e18/tasks/base.py#L293C1-L294C1) so it is actually $y-x$ as you mentioned. Sorry for the confusion.

**Comment by zhxieml** [comment]
Thank you for your prompt reply!

I'm still confused thought. What does the opposite sign represent? If I understand correctly, in [get_icv](https://github.com/shengliu66/ICV/blob/ae4fb0a2bbbb4e468053e9c357a42efda4990e18/tasks/base.py#L293), `hidden_states[demonstration_id][0].flatten()` represents $x$, and `hidden_states[demonstration_id][1].flatten()` represents $y$. Are we still subtracting $x - y$ here?

**Comment by shengliu66** [comment]
In the get_icv we compute $x-y$, and in the notebook, we subtract $x-y$, thus is we add $y-x$ which is consistent with the paper. Hope this explanation is clear.

**Comment by zhxieml** [comment]
> In the get_icv we compute x−y, and in the notebook, we subtract x−y, thus is we add y−x which is consistent with the paper. Hope this explanation is clear.

Oh, I just realized that when applying icvs, it involves subtraction (`icv_all_tasks -= alpha[i] * lambda_sim * F.normalize(self.icvs[i], dim=-1).repeat(1,x.shape[1],1)`). That would explain everything. Thank you.

I have another question regarding normalization though. Before PCA, it seems like [the differences are not normalized](https://github.com/shengliu66/ICV/blob/ae4fb0a2bbbb4e468053e9c357a42efda4990e18/tasks/base.py#L293). Does it contradict with the paper (Sec 3.1)?

**Comment by RadiantCrystal** [comment]
> > In the get_icv we compute x−y, and in the notebook, we subtract x−y, thus is we add y−x which is consistent with the paper. Hope this explanation is clear.
> 
> Oh, I just realized that when applying icvs, it involves subtraction (`icv_all_tasks -= alpha[i] * lambda_sim * F.normalize(self.icvs[i], dim=-1).repeat(1,x.shape[1],1)`). That would explain everything. Thank you.
> 
> I have another question regarding normalization though. Before PCA, it seems like [the differences are not normalized](https://github.com/shengliu66/ICV/blob/ae4fb0a2bbbb4e468053e9c357a42efda4990e18/tasks/base.py#L293). Does it contradict with the paper (Sec 3.1)?

Hello Authors, Can you please answer the last question mentioned above? Also can you clearly tell how the direction of the incontext vector has been considered in your demo_notebook.ipynb file? Thanks.
